docs/.hooks/inject_version.py

```diff
@@ -23,9 +26,9 @@ def get_latest_version():
     return f'{major}.{minor}.{patch}'
 
 
-def on_page_read_source(
-    page,
-    config,  # noqa: ARG001
-):
-    with open(page.file.abs_src_path, encoding='utf-8') as f:
-        return f.read().replace(MARKER, get_latest_version())
+class VersionInjectionPreprocessor(Preprocessor):
```

Comment:
Also FYI for a bit more context:

These 3 mechanisms are all equivalent but with the following differences:

* on_page_read_content is the least flexible and has some problems https://github.com/mkdocs/mkdocs/issues/3486#issuecomment-1827136857
* on_page_markdown just gives you the source directly and just works
* Adding a markdown preprocessor - still kinda equivalent but more deeply connected to the pipeline, thus letting it also apply even inside docstrings with mkdocstrings, this being the reason why I had chosen that approach for the hook that I contributed

Reply:
Oh that preprocessor bit is interesting, I think I'll keep it this way in case I want to do that in future